ports/atmel-samd/supervisor/port.c

```diff
@@ -167,21 +167,24 @@ safe_mode_t port_init(void) {
 
 void reset_port(void) {
     // Reset all SERCOMs except the one being used by the SPI flash.
-//     Sercom *sercom_instances[SERCOM_INST_NUM] = SERCOM_INSTS;
-//     for (int i = 0; i < SERCOM_INST_NUM; i++) {
-//         #ifdef SPI_FLASH_SERCOM
-//             if (sercom_instances[i] == SPI_FLASH_SERCOM) {
-//                 continue;
-//             }
-//         #endif
-//         #ifdef MICROPY_HW_APA102_SERCOM
-//             if (sercom_instances[i] == MICROPY_HW_APA102_SERCOM) {
-//                 continue;
-//             }
-//         #endif
-//         sercom_instances[i]->SPI.CTRLA.bit.SWRST = 1;
-//     }
-//
+    Sercom *sercom_instances[SERCOM_INST_NUM] = SERCOM_INSTS;
+    for (int i = 0; i < SERCOM_INST_NUM; i++) {
+#ifdef SPI_FLASH_SERCOM
+        if (sercom_instances[i] == SPI_FLASH_SERCOM) {
+            continue;
+        }
+#endif
+#ifdef MICROPY_HW_APA102_SERCOM
+        if (sercom_instances[i] == MICROPY_HW_APA102_SERCOM) {
+            continue;
+        }
+#endif
+        sercom_instances[i]->SPI.CTRLA.bit.SWRST = 1;
+        sercom_instances[i]->I2CM.CTRLA.bit.ENABLE = 0;
+        sercom_instances[i]->I2CM.CTRLA.bit.SWRST = 1;
```

Comment:
I don't think you need these three things. SPI.CTRLA.bit.SWRST is the same memory location as I2CM.CTRLA.bit.SWRST because SPI and I2CM are members of a union. Instead, I think you had "in use" errors because the reset takes time and the M4 is faster than the M0. You can wait on the state of SYNCBUSY.bit.SWRST. It'll be 0 when the reset is finished.

Reply:
Union thing understood. I didn't read the register defns carefully enough.
The weird thing about the ENABLE and SWRST is that I got "pin in use" errors after a soft reboot when lines 183 and 184 were reversed. These errors were many seconds after the reboot: I was testing in the REPL. I'll test this again.

Reply:
I thought maybe SWRST was ENABLE-protected. It says it isn't.